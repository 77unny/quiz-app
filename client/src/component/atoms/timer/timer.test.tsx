import 'jest-styled-components';

import { render } from '@testing-library/react';
import React from 'react';
import { act } from 'react-dom/test-utils';

import Timer from './index';

beforeEach(() => {
  jest.useFakeTimers();
});

afterEach(() => {
  jest.useRealTimers();
  jest.clearAllMocks();
  jest.useRealTimers();
});

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));

describe('<Timer />', () => {
  it('[Timer] prefix - 지속시간', () => {
    const { container } = render(<Timer prefix={'지속시간'} />);

    expect(container.firstChild).toMatchSnapshot(`지속시간`);
  });

  it('[Timer] 1초 경과  - 00:01', () => {
    const { getByText } = render(<Timer />);
    act(() => {
      jest.advanceTimersByTime(1000);
    });

    expect(getByText('00:01'));
  });

  it('[Timer] 61초 경과  - 01:01', () => {
    const { getByText } = render(<Timer />);
    act(() => {
      jest.advanceTimersByTime(61000);
    });

    expect(getByText('01:01'));
  });
});
