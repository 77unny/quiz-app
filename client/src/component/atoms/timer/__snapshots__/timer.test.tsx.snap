// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timer /> [Timer] prefix - 지속시간: 지속시간 1`] = `
.c0 {
  font-weight: 300;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-style: italic;
}

<p
  class="c0"
>
  <span>
    지속시간
     
  </span>
  <span>
    00:00
  </span>
</p>
`;
